packages/core/src/controllers/subscriber.ts

```diff
@@ -384,6 +384,7 @@ export class Subscriber extends ISubscriber {
   private async batchSubscribe(subscriptions: SubscriberTypes.Params[]) {
     if (!subscriptions.length) return;
     const result = (await this.rpcBatchSubscribe(subscriptions)) as string[];
+    if (!result) return;
```

Comment:
Empty array `[]` is still truthy so I think this is a fall through, better to go by `.length` assuming `rpcBatchSubscribe` will always return an array.
```suggestion
    if (!result.length) return;
```

Reply:
If the connection is down i.e. wifi switched off, the batch subscribe will resolve with `undefined` after the request timeouts. I will also add `.length` to be on the safe side 💯